src/storage/mapping/ExtensionBasedMapperFactory.ts

```diff
@@ -0,0 +1,8 @@
+import { ExtensionBasedMapper } from './ExtensionBasedMapper';
+import type { FileIdentifierMapperFactory } from './FileIdentifierMapperFactory';
+
+export class ExtensionBasedMapperFactory implements FileIdentifierMapperFactory<ExtensionBasedMapper> {
```

Comment:
Maybe add TSdoc as to why this is needed?
My guess is that you want one per pod.

BTW I'd also be fine putting this class into the `ExtensionBasedMapper` file.